ext/storage_sources/azure_store/azure_storage_source.cpp

```diff
@@ -181,32 +183,71 @@ azure_add_reference(WT_STORAGE_SOURCE *storage_source)
     return 0;
 }
 
+// Flush a file locally to Azure.
 static int
 azure_flush(WT_STORAGE_SOURCE *storage_source, WT_SESSION *session, WT_FILE_SYSTEM *file_system,
   const char *source, const char *object, const char *config)
 {
     WT_UNUSED(storage_source);
-    WT_UNUSED(session);
-    WT_UNUSED(file_system);
-    WT_UNUSED(source);
-    WT_UNUSED(object);
     WT_UNUSED(config);
+    azure_file_system *azure_fs = reinterpret_cast<azure_file_system *>(file_system);
+    WT_FILE_SYSTEM *wtFileSystem = azure_fs->wt_fs;
 
-    return 0;
+    if (!std::filesystem::exists(object)) {
```

Comment:
Hmm do we need this here since we are checking down below instead?

Reply:
yeah i tried to do this but filesystem::canonical will complain if the file doesn't exist. Yes its redundancy since its the same check twice essentially

Reply:
Oh thats unfortunate can we add a comment to explain the redundancy?